test/api/python/provisioner/test_inputs.py

```diff
@@ -744,3 +746,152 @@ def _check(*args):
     # urls
     res = SWUpdateRepo(some_release, source='https://some/http/url')
     _check('is_remote')
+
+
+def test_inputs_copy_attr_verify_attrs():
+    some_cls = attr.make_class("some_cls", {
+        "some_attr": attr.ib(
+            default=123,
+            validator=None,
+            metadata={
+                'metadata_key': 1
+            },
+            repr=False,
+            hash=None,
+            init=True,
+            type=str,
+            converter=None,
+            kw_only=True
+        )
+    })
+
+    read_attr = attr.fields(some_cls).some_attr
+
+    copied_attr = inputs.copy_attr(read_attr)
+
+    assert copied_attr == read_attr
+
+
+def test_ParserFiller_fill_parser_when_action_is_store_bool():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            metadata={
+                METADATA_ARGPARSER: {
+                    'action': 'store_bool',
+                    'help': 'some help'
+                }
+            },
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
+    args = parser.parse_args(['--y', '--noy'])
+    assert not args.y
+
+    args = parser.parse_args(['--y'])
+    assert args.y
+
+    args = parser.parse_args(['--noy'])
+    assert not args.y
+
+
+def test_ParserFiller_fill_parser_with_no_action_in_metadata():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            metadata={
+                METADATA_ARGPARSER: {
+                    'help': 'some help'
+                }
+            },
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
+    args = parser.parse_args(['--y', 'some-value'])
+    assert args.y == 'some-value'
+
+
+def test_ParserFiller_fill_parser_with_no_metadata_argparser():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
```

Comment:
you don't do any checks here, what is the purpose of that test?

Reply:
It was a suggestion given in the old PR : https://github.com/Seagate/cortx-prvsnr/pull/338/commits/f039adba116fb689a81dd706a99cd9f810a1d45c
ParserFiller is not covered but as long as it just passes result of the AttrParserArgs to argparse we should focus on special cases:

- a class has some argument that doesn't specify METADATA_ARGPARSER - that argument should be ignored, so calls to parser.parse_args should produce an error

This unit test is implemented to cover above test scenario, but it doesn't throw any error.
Please suggest further, whether should we keep it or remove ?

Reply:
>so calls to parser.parse_args should produce an error
but you don't do that here

Reply:
working on it

Reply:
When I am calling parser.parse_args(), it is throwing below error:
`>           args = parser.parse_args(['--y', 'some-value'])

test/api/python/provisioner/test_inputs.py:828:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.7/argparse.py:1758: in parse_args
    self.error(msg % ' '.join(argv))
/usr/local/lib/python3.7/argparse.py:2508: in error
    self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = ArgumentParser(prog='pytest', usage=None, description=None, formatter_class=<class 'argparse.HelpFormatter'>, conflict_handler='error', add_help=True)
status = 2, message = 'pytest: error: unrecognized arguments: --y some-value\n'

    def exit(self, status=0, message=None):
        if message:
            self._print_message(message, _sys.stderr)
>       _sys.exit(status)
E       SystemExit: 2

/usr/local/lib/python3.7/argparse.py:2495: SystemExit
`
It is not raising any exception as such in order to catch within pytest.raises.
Should I mark this test with pytest  marker as @pytest.mark.xfail ?

Reply:
@sudarshana27 yes, that is what we should expect, try the following:
```
    with pytest.raises(SystemExit) as excinfo:                                                                                                                               
       args = parser.parse_args(['--y', 'some-value'])
```
you may also check `unrecognized arguments: --y some-value'` in `excinfo`

Reply:
Addressed